lighthouse-core/gather/gatherers/dobetterweb/domstats.js

```diff
@@ -55,7 +57,11 @@ function elementPathInDOM(element) {
   const path = [createSelectorsLabel(element)];
   let node = element;
   while (node) {
-    node = node.parentNode && node.parentNode.host ? node.parentNode.host : node.parentElement;
+    // Anchor elements have a .host property. Be sure we've found a shadow root
+    // host and not an anchor.
+    const shadowHost = node.parentNode && node.parentNode.host &&
```

Comment:
This is it :( https://developers.google.com/web/fundamentals/getting-started/primers/shadowdom?hl=en#create

Typically, you don't traverse upwards through the entire deep DOM tree like this. So it's a big of a special case for what we're doing.